lib/services/serviceBus/lib/notificationhubservice.js

```diff
@@ -343,19 +343,19 @@ NotificationHubService.prototype.createOrUpdateRegistration = function (registra
       delete registration.Expiry;
   }
 
-  if (registration.RegistrationId != null) {
+  if (registration.RegistrationId !== null || registration.RegistrationId !== undefined) {
       delete registration.RegistrationId;
   }
 
     var newRegistration = {};
     newRegistration.Tags = registration.Tags;
     for (var key in registration)
     {
-        if(key!="Tags")
+        if(key !== 'Tags')
         newRegistration[key] = registration[key];
     }
 
-    if(registration.BodyTemplate!=null)
+    if(registration.BodyTemplate !== null || registration.BodyTemplate !== undefined)
```

Comment:
This should have a **logical AND**. Basically we want to check that if registration object has BodyTemplate property then we want to do something with newRegistration. Based on the current logic, even when registration.BodyTemplate is null or undefined, the if condition will pass.